pkg/curatedpackages/packagecontrollerclient_test.go

```diff
@@ -54,14 +75,52 @@ func TestInstallControllerSuccess(t *testing.T) {
 	registry := curatedpackages.GetRegistry(tt.ociUri)
 	sourceRegistry := fmt.Sprintf("sourceRegistry=%s", registry)
 	values := []string{sourceRegistry}
+	params := []string{"create", "-f", "-", "--kubeconfig", tt.kubeConfig}
+	templateValues := map[string]string{
+		"eksaAccessKeyId":     "",
+		"eksaSecretAccessKey": "",
+		"eksaRegion":          base64.StdEncoding.EncodeToString([]byte(eksaDefaultRegion)),
+	}
+	result, err := templater.Execute(awsSecretYaml, templateValues)
+	tt.Expect(err).To(BeNil())
+	tt.kubectl.EXPECT().CreateFromYaml(tt.ctx, result, params).Return(bytes.Buffer{}, nil)
+	params = []string{"create", "job", jobName, "--from=" + cronJobName, "--kubeconfig", tt.kubeConfig, "--namespace", constants.EksaPackagesName}
+	tt.kubectl.EXPECT().ExecuteCommand(tt.ctx, params).Return(bytes.Buffer{}, nil)
 	tt.chartInstaller.EXPECT().InstallChart(tt.ctx, tt.chartName, "oci://"+tt.ociUri, tt.chartVersion, tt.kubeConfig, values).Return(nil)
 
-	err := tt.command.InstallController(tt.ctx)
+	err = tt.command.InstallController(tt.ctx)
 	if err != nil {
 		t.Errorf("Install Controller Should succeed when installation passes")
 	}
 }
 
+func TestInstallControllerSuccessWhenRegionSpecified(t *testing.T) {
+	tt := newPackageControllerTest(t)
+
+	registry := curatedpackages.GetRegistry(tt.ociUri)
+	sourceRegistry := fmt.Sprintf("sourceRegistry=%s", registry)
+	values := []string{sourceRegistry}
+	params := []string{"create", "-f", "-", "--kubeconfig", tt.kubeConfig}
+	templateValues := map[string]string{
+		"eksaAccessKeyId":     "",
+		"eksaSecretAccessKey": "",
+		"eksaRegion":          base64.StdEncoding.EncodeToString([]byte(tt.eksaRegion)),
+	}
+	os.Setenv(config.EksaRegionEnv, tt.eksaRegion)
```

Comment:
You can make use of https://pkg.go.dev/testing#T.Setenv now to not have to worry about cleaning it up at the end.